src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/Dumpling.targets

```diff
@@ -1,25 +1,30 @@
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup>
-    <Content Include="$(MSBuildThisFileDirectory)Dumpling.sh">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
     <Content Include="$(MSBuildThisFileDirectory)DumplingHelper.py">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
+    <DumplingIncPaths Include="$(RuntimePath)" />
+    <DumplingIncPaths Include="$(TestPath)" />
   </ItemGroup>
+  <Target Name="PreinstallDumpling"
+          BeforeTargets="TestAllProjects">
+    <Exec Command="python $(MSBuildThisFileDirectory)DumplingHelper.py install_dumpling" />
+  </Target>
   <!-- Setup Dumpling service to collect crash dumps -->
   <Target Name="SetupDumpling"
           BeforeTargets="GenerateTestExecutionScripts">
+    <PropertyGroup>
+      <__DumplingIncPathsArg>@(DumplingIncPaths -> '%(Identity)', ',')</__DumplingIncPathsArg>
+    </PropertyGroup>
     <ItemGroup Condition="'$(TargetOS)'!='Windows_NT'">
-      <TestCommandLines Include="source Dumpling.sh" />
-      <PostExecutionTestCommandLines Include="CollectDumps $%3f $(MSBuildProjectName)" />
+      <TestCommandLines Include="__TIMESTAMP=`python DumplingHelper.py get_timestamp`" />
+      <PostExecutionTestCommandLines Include="python DumplingHelper.py collect_dump $%3F `pwd` $__TIMESTAMP $(MSBuildProjectName) $(__DumplingIncPathsArg)" />
     </ItemGroup>
     <Error Condition="'$(TargetOS)' == 'Windows_NT' And '$(CrashDumpFolder)' == ''" Text="CrashDumpFolder must be set to use Dumpling on Windows." />
     <ItemGroup Condition="'$(TargetOS)'=='Windows_NT'">
-      <TestCommandLines Include="python DumplingHelper.py install_dumpling" />
```

Comment:
Matt pointed out that, as long as the pre-install step runs before this, then it doesn't matter if a bunch of tests are trying to install in parallel (because they exit early when the file already exists). Helix doesn't run tests in parallel so it won't be a problem there either. Also, devs running individual tests with EnableDumpling=true will be able to do so without running install_dumpling first.